### ISSUE TITLE
Using go-rules with please v16

### ISSUE THREAD

**Issue by boolangery** [please-build/go-rules#321]
We're using this plugin with Please v16.28.0 (because there is too much work to migrate to 17 for now)

So far it's working with this warning:
```
WARNING: Plugin "go" requires Please version 17.11.0
```

Why does this plugin require Please v17 if it seems to work with v16? What are the possible side effects of using it with an older version?

**Comment by Tatskaari** [comment]
I think mostly we're using features of the new build system that may not be available in v16. Core functionality may work, but you may find that as your usage grows you'll hit code paths that depend on newer features of Please.
